src/metabase/models/interface.clj

```diff
@@ -335,9 +393,16 @@
   *  `(partial current-user-has-full-permissions? :write)` (you must also implement [[perms-objects-set]] to use this)
   *  `(partial current-user-has-partial-permissions? :write)` (you must also implement [[perms-objects-set]] to use
       this)"
-  {:arglists '([instance] [model pk]), :hydrate :can_write}
+  {:arglists '([instance] [model pk])}
   dispatch-on-model)
 
+#_{:clj-kondo/ignore [:unused-private-var]}
+(define-simple-hydration-method ^:private hydrate-can-write
+  :can_write
+  "Hydration method for `:can_write`."
+  [instance]
+  (can-write? instance))
```

Comment:
Had to create a dummy function here because `can-write?` is a `defmulti`... with Toucan 2 we could maybe eliminate the multimethod entirely since hydration can dispatch off of both model and hydration key